pkg/controllers/subnet/subnet_controller.go

```diff
@@ -242,5 +244,58 @@ func (r *SubnetReconciler) Start(mgr ctrl.Manager) error {
 	if err != nil {
 		return err
 	}
+	go r.GarbageCollector(make(chan bool), servicecommon.GCInterval)
 	return nil
 }
+
+func (r *SubnetReconciler) GarbageCollector(cancel chan bool, timeout time.Duration) {
+	ctx := context.Background()
+	log.Info("subnet garbage collector started")
+	for {
+		select {
+		case <-cancel:
+			return
+		case <-time.After(timeout):
+		}
+		nsxSubnetList := r.Service.ListSubnet()
+		if len(nsxSubnetList) == 0 {
+			continue
+		}
+
+		crdSubnetList := &v1alpha1.SubnetList{}
+		err := r.Client.List(ctx, crdSubnetList)
+		if err != nil {
+			log.Error(err, "failed to list subnet CR")
+			continue
+		}
+
+		crdSubnetIDs := sets.NewString()
+		for _, sr := range crdSubnetList.Items {
+			crdSubnetIDs.Insert(string(sr.UID))
+		}
+
+		crdSubnetSetIDs := r.Service.ListSubnetSetID(ctx)
```

Comment:
This should be put in subnetset controller?

Reply:
both subnet/subnetset controller have the logic to GC the subnet.
They will have lots of duplicate code.
subnet controller also needs to check if subnet belongs to subnetset.
subnetset controller needs to list all subnet.
Only one place to handle subnet GC will make it simple.